### PR TITLE
[4.0] Update libraries from Composer

### PR DESCRIPTION
**Pull request by twister65**
Pull Request for Issue # .

### Summary of Changes
`composer update --no-dev`
This will update all Composer files to the latest versions as allowed in the composer.json file, except for the **application**, **oauth1**, **oauth2**, and **console** package.
This includes the Joomla! Framework packages !

### Testing Instructions
Get a working installation from the source:

- Clone the repository:
`git clone git@github.com:joomla/joomla-cms.git`

- Go to the joomla-cms folder:
`cd joomla-cms`

- Go to the 4.0-dev branch:
`git checkout 4.0-dev`

- Install all the needed composer packages:
`composer install`

- Install all the needed npm packages:
`npm install`

### CODE REVIEW

**Comment by wilsonge** [comment]
I suspect this is going to break the Joomla console package. similar to how my version of this a while back did? https://github.com/joomla/joomla-cms/pull/26820 

We can probably selectively update other parts whilst I try and get a plan around the console package though

**Comment by twister65** [comment]
@wilsonge , I removed the update for the console package.

**Comment by mbabker** [comment]
See #27997 - Everything but the application and two OAuth packages are safe to update.

(Also, if you take out an update by manually editing the `composer.lock`, you need to run `composer update --lock` otherwise you are going to cause everyone to get warnings about the lock file hash being incorrect)

**Comment by twister65** [comment]
@mbabker , if we update the console package, we still get this error:
console_error
I suggest not updating the joomla console package until @wilsonge finds a solution. This will be the subject of another PR.

**Comment by mbabker** [comment]
Bah, forgot that change.

Either way, you’re still introducing the Composer lock file hash error as the content-hash property isn’t updated here.

**Comment by twister65** [comment]
@mbabker , when I run `composer update --lock`, the `composer.lock` file is always the same. I may have forgotten something.

**Comment by mbabker** [comment]
Something is definitely wrong though because it should be impossible to have run `composer update` and not change the hash, yet if you look at the diff for this PR the hash is unchanged.  That’s a red flag.

**Comment by twister65** [comment]
> This will update all Composer files to the latest versions as allowed in the composer.json file, except for the application, oauth1, oauth2, and console package.

composer

File: composer.lock [hunk 1]
```diff
@@ -394,17 +397,17 @@
             },
             "require": {
                 "joomla/event": "~2.0",
-                "joomla/input": "~1.2|~2.0",
                 "joomla/registry": "^1.4.5|~2.0",
-                "php": "~7.2",
+                "laminas/laminas-diactoros": "^2.2.2",
```
Comment (mbabker) [inline composer.lock:401]: You might need to just redo this PR from scratch.  The Zend to Laminas changes shouldn't be in here if you did the `composer update` steps correctly.  This to me implies that the lock file has been manually modified to reach your desired end state, and this is going to cause a lot of problems.

File: composer.lock [hunk 2]
```diff
@@ -564,8 +568,8 @@
                 "joomla/application": "~2.0",
                 "joomla/event": "~2.0",
                 "joomla/string": "~2.0",
-                "php": "~7.2",
-                "symfony/console": "~3.4|~4.0"
+                "php": "^7.2.5",
+                "symfony/console": "~3.4|~4.0|~5.0"
```
Comment (mbabker) [inline composer.lock:572]: Same here, if you reverted the update of `joomla/console` correctly this change wouldn't be present.

Comment (twister65) [reply composer.lock:572]: Thanks, I will do an interactive update.

**Pull request closed without merging**